src/modules/helpers/index.ts

```
@@ -245,6 +245,302 @@ export class HelpersModule {
+   * faker.helpers.fromRegExp('[a-d0-6]{2,8}') // 'a0dc45b0'
+   * faker.helpers.fromRegExp('[-a-z]{5}') // 'a-zab'
+   * faker.helpers.fromRegExp(/[A-Z0-9]{4}-[A-Z0-9]{4}/) // 'BS4G-485H'
+   * faker.helpers.fromRegExp(/[A-Z]{5}/i) // 'pDKfh'
+   * faker.helpers.fromRegExp(/.{5}/) // '14(#B'
+   * faker.helpers.fromRegExp(/Joh?n/) // 'Jon'
+   * faker.helpers.fromRegExp(/ABC*DE/) // 'ABDE'
+   * faker.helpers.fromRegExp(/bee+p/) // 'beeeeeeeep'
+   *
+   * @since 8.0.0
+   */
+  fromRegExp(pattern: string | RegExp): string {
+    let isCaseInsensitive = false;
+
+    if (pattern instanceof RegExp) {
+      isCaseInsensitive = pattern.flags.includes('i');
+      pattern = pattern.toString();
+      pattern = pattern.match(/\/(.+?)\//)?.[1] ?? ''; // Remove frontslash from front and back of RegExp
+    }
+
+    let min: number;
+    let max: number;
+    let repetitions: number;
+
+    // Deal with single wildcards
+    const SINGLE_CHAR_REG =
+      /([.A-Za-z0-9])(?:\{(\d+)(?:\,(\d+)|)\}|(\?|\*|\+))(?![^[]*]|[^{]*})/;
+    let token = pattern.match(SINGLE_CHAR_REG);
+    while (token != null) {
+      const quantifierMin: string = token[2];
+      const quantifierMax: string = token[3];
+      const quantifierSymbol: string = token[4];
+
+      if (quantifierSymbol) {
+        switch (quantifierSymbol) {
+          case '?': {
+            repetitions = this.faker.datatype.boolean() ? 0 : 1;
+            break;
+          }
+
+          case '*': {
+            repetitions = 0;
+            while (this.faker.datatype.boolean(1 / (repetitions + 1))) {
+              repetitions++;
+            }
+
+            break;
+          }
+
+          case '+': {
+            repetitions = 1;
+            while (this.faker.datatype.boolean(1 / repetitions)) {
+              repetitions++;
+            }
+
+            break;
+          }
+
+          // Technically would never happen
+          default:
+            repetitions = 1;
```

Then lets throw an error here.